toolkit/docs/how_it_works/0_intro.md

```diff
@@ -2,23 +2,308 @@ How The Build System Works
 ===
 ## Next: [Initial Prep](1_initial_prep.md)
 ## General Build Flow
-Graph showing how the various build components interact.
 
-Edges are marked with the tool used in blue when applicable.
+This section is intended to give an overview of Mariner's build process and toolkit. The diagrams below follow conventions stated in the key.
+```mermaid
+---
+title: Flowchart key
+---
+flowchart TD
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
 
-![Build Flow Graph](images/flow.png)
+    %% node definitions
+    input[/input or output/]:::io
+    process[process]:::process
+    decision{{decision}}:::decision
+    goodstate([good state]):::goodState
+    badstate[[bad state]]:::badState
+```
+<br/>
 
-The build process can be split into three components: tooling, package generation, and image generation.
+### High-level RPM Flow
+Mariner is an RPM based distro. A single package (or RPM) is built using a combination of sources and a spec file. A signature file is used to verify the sources' hashes. 
+
+```mermaid
+flowchart LR
+    %% style definitions
+    classDef io fill:#247BA0,stroke:#333,stroke-width:2px,color:#fff;
+    classDef process fill:#B05E2F,stroke:#333,stroke-width:2px,color:#fff;
+    classDef decision fill:#51344D,stroke:#333,stroke-width:2px,color:#fff;
+    classDef goodState fill:#566E40,stroke:#333,stroke-width:2px,color:#fff;
+    classDef badState fill:#BC4B51,stroke:#333,stroke-width:2px,color:#fff;
+
+    %% nodes
+    spec[/Spec/]:::io
+    localSourceTar[/Local Sources if present/]:::io
+    sigFile[/Signature file/]:::io
+    remoteSourceTar[/Remote Source/]:::io
+    patches[/Patches/]:::io
+    pack[Pack SRPM]:::process
+    srpm[/SRPM/]:::io
+    buildRPM[Build RPM]:::process
+    rpm[/RPM/]:::io
+
+    %% node flow
+    spec --> pack
+    localSourceTar --> pack
+    remoteSourceTar --> pack
+    patches --> pack
+    sigFile --> pack
+    pack -->srpm 
+    srpm --> buildRPM
+    buildRPM --> rpm
+```
+</br>
+
+### High-level Build Flow
+ The build process can be split into three components: tooling, package generation, and image generation. When building, Makefile options can be used to build Mariner entirely from end to end or to download prebuilt artifacts.
```

Comment:
Technically, the tool is called "make" and a file called "Makefile" is its default entry point. What do you think about this:
```suggestion
 The build process can be split into three components: tooling, package generation, and image generation. When building, `make` options can be used to build Mariner entirely from end to end or to download prebuilt artifacts.
```